betty/plugins/anonymizer/__init__.py

```diff
@@ -8,7 +8,34 @@
 from betty.plugins.privatizer import Privatizer
 
 
+class AnonymousSource(Source):
+    def __init__(self):
+        Source.__init__(self, _('Private'))
+
+    def replace(self, other: Source) -> None:
+        self.citations.append(*other.citations)
+        self.contained_by = other.contained_by
+        self.contains.append(*other.contains)
+        self.files.append(*other.files)
+
+
+class AnonymousCitation(Citation):
+    def __init__(self, source: Source):
+        Citation.__init__(self, source)
+
+    @property
+    def location(self) -> Optional[str]:
+        return _("This citation has not been published in order to protect people's privacy")
+
+    def replace(self, other: Citation) -> None:
+        self.facts.append(*other.facts)
+        self.files.append(*other.files)
+        self.source = other.source
```

Comment:
This line must be removed, as the containing source is *always* an `AnonymousSource`.